x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/api/rules/import_rules/route.ts

```diff
@@ -138,34 +146,66 @@ export const importRulesRoute = (router: SecuritySolutionPluginRouter, config: C
 
           // rulesWithMigratedActions: Is returned only in case connectors were exported from different namespace and the
           // original rules actions' ids were replaced with new destinationIds
-          const parsedRules = actionConnectorErrors.length
+          const parsedRuleStream = actionConnectorErrors.length
             ? []
             : rulesWithMigratedActions || migratedParsedObjectsWithoutDuplicateErrors;
 
-          const chunkParseObjects = chunk(CHUNK_PARSED_OBJECT_SIZE, parsedRules);
-
-          const importRuleResponse: ImportRuleResponse[] = await importRulesHelper({
-            ruleChunks: chunkParseObjects,
-            rulesResponseAcc: [...actionConnectorErrors, ...duplicateIdErrors],
-            overwriteRules: request.query.overwrite,
-            detectionRulesClient,
-            allowMissingConnectorSecrets: !!actionConnectors.length,
-            savedObjectsClient,
+          const ruleSourceImporter = createRuleSourceImporter({
```

Comment:
@banderror I tried to add a lazy, memoized factory for this to the request context, but it wasn't straightforward  since the importer itself depends on the context itself. Ultimately it just needs `getFleetServices` and `appClient`, but refactoring the functions involved in `installPrebuiltRulesPackage` didn't seem worth it for a class that's really only needed during import. If we end up needing it again elsewhere, we can revisit.

Reply:
@rylnd ACK, I'll do my best to review the PR this Friday.

Reply:
@rylnd Yeah, I see that. Refactoring `installPrebuiltRulesPackage` and other related functions would require some effort that is out of the scope of this PR and the whole work on importing prebuilt rules. I think we can refactor this area of code later, after the completion of Milestone 3.